cmd/whitesourceExecuteScan.go

```diff
@@ -245,34 +258,37 @@ func checkSecurityViolations(config *ScanOptions, sys *System) error {
 
 // pollProjectStatus polls project LastUpdateTime until it reflects the most recent scan
 func pollProjectStatus(config *ScanOptions, sys *System) error {
-	currentTime := time.Now()
+	log.Entry().Info("Polling project status before downloading reports...")
+	log.Entry().Infof("Project token: %s", config.ProjectToken)
+	retry := 30
 	for {
-		project, err := sys.GetProjectVitals(config.ProjectToken)
+		reportBytes, err := sys.GetProjectRiskReport(config.ProjectToken)
 		if err != nil {
 			return err
 		}
 
-		// Make sure the project was updated in whitesource backend before downloading any reports
-		lastUpdatedTime, err := time.Parse("2006-01-02 15:04:05 +0000", project.LastUpdateDate)
-		if currentTime.Sub(lastUpdatedTime) < 10*time.Second {
-			//done polling
+		if len(reportBytes) > 36000 {
```

Comment:
how can we be sure with `>36000`?

Would it make sense to perform the last update check in addition? Do we know that it is the latest content in the report?

Reply:
I was using `lastUpdatedTime` in a previous attempt.. The problem was the `lastUpdatedTime` of the project is updated before the actual PDF risk report was "ready" for download. The average "blank" pdf report size ranged from 34,000-35,000 bytes depending on the length of the project name. Once the file size goes above 36,000 I think it's safe to say there were at least a few detected libraries...

It is still definitely not 100% accurate obviously and I'm sure needs further testing...

Reply:
would it make sense to retrieve `lastUpdatedTime` twice, first time before WhiteSource execution, second time afterwards?

Maybe in addition to the report size?

Reply:
@nevskrem, @CCFenner any other ideas here?

Reply:
@OliverNocon Yes, that would make much more sense than the current implementation of checking if the lastUpdatedTime is "recent" (within 10 seconds of the scan finishing).